crates/web5/src/test_vectors.rs

```diff
@@ -207,4 +208,100 @@ mod test_vectors {
             }
         }
     }
+
+    mod crypto_ed25519 {
+        use super::*;
+        use crate::crypto::dsa::ed25519::{Ed25519Signer, Ed25519Verifier};
+        use crate::crypto::dsa::{Signer, Verifier};
+        use crate::crypto::jwk::Jwk;
+
+        #[derive(Debug, serde::Deserialize)]
+        struct SignVectorInput {
+            data: String,
+            key: Jwk,
+        }
+
+        #[derive(Debug, serde::Deserialize)]
+        struct VerifyVectorInput {
+            data: String,
+            key: Jwk,
+            signature: String,
+        }
+
+        #[test]
+        fn sign() {
+            let path = "crypto_ed25519/sign.json";
+            let vectors: TestVectorFile<SignVectorInput, Option<String>> =
+                TestVectorFile::load_from_path(path);
+
+            for vector in vectors.vectors {
+                let input = vector.input;
+                let expected_output = vector.output;
+
+                let signer = Ed25519Signer::new(input.key);
+
+                let data = hex_string_to_byte_array(input.data.to_string());
+                let result = signer.sign(&data);
+
+                if matches!(vector.errors, Some(true)) {
+                    assert!(result.is_err(), "Expected an error, but signing succeeded");
+                } else {
+                    let signature = result.expect("Signing should not fail");
+
+                    // Convert the signature to a hex string
+                    let signature_hex = byte_array_to_hex_string(&signature);
+
+                    assert_eq!(
+                        signature_hex,
+                        expected_output.unwrap(),
+                        "Signature does not match expected output"
+                    );
+                }
+            }
+        }
+
+        #[test]
+        fn verify() {
+            let path = "crypto_ed25519/verify.json";
+            let vectors: TestVectorFile<VerifyVectorInput, bool> =
+                TestVectorFile::load_from_path(path);
+
+            for vector in vectors.vectors {
+                let input = vector.input;
+                let expected_output = vector.output;
+
+                let verifier = Ed25519Verifier::new(input.key);
+
+                let data = hex_string_to_byte_array(input.data.to_string());
+                let signature = hex_string_to_byte_array(input.signature.to_string());
+
+                let result = verifier.verify(&data, &signature);
+
+                let is_valid = result.expect("Verification should not fail");
```

Comment:
what's the difference between Err() and Ok(false) for `verifier.verify`?

Reply:
chad says:
Ok(false): The operation itself succeeded, but the verification result is false. This means that the signature is valid in terms of format and process, but it does not match the data or is otherwise invalid according to the verification logic.

and yea then err is an exception basically if I'm understnading rust correctly

Reply:
oh and as far as the implementation yea just verified false is invalid signature vs catastrophic failure

Reply:
FYI in #300 I'm moving in favor of the monad pattern so `verify` returns a `void` and utilizes the error case for cryptographic failure 

no actionable items in this PR, just making you aware (and I'll be sure to remediate test vectors prior to merging #300 )